File: app/src/main/java/com/kakacat/minitool/textencryption/Presenter.java
package com.kakacat.minitool.textencryption;

import android.text.TextUtils;

public class Presenter implements Contract.Presenter {

    private Contract.View view;
    private Model model;

    public Presenter(Contract.View view) {
        this.view = view;
        this.model = Model.getInstance();
    }

    @Override
    public void initData() {
    }

    @Override
    public String[] getEncryptionMethods() {
        return model.getEncryptionMethods();
    }

    @Override
    public void encryptText(String content, CharSequence method) {
        if (TextUtils.isEmpty(method)) {
            view.onEncryptResult(null);
        } else {
            view.onEncryptResult(model.encryptText(content, method));
        }
    }
}

File: app/src/main/java/com/kakacat/minitool/textencryption/Presenter.kt
package com.kakacat.minitool.textencryption

import android.text.TextUtils

class Presenter(private val view: Contract.View) : Contract.Presenter {

    private val model by lazy { Model() }

    override fun initData() {}

    override val encryptionMethods: Array<String>
        get() = Model.encryptionMethods

    override fun encryptText(content: String?, method: CharSequence?) {
        if (TextUtils.isEmpty(method)) {
            view.onEncryptResult(null)
        } else {
            view.onEncryptResult(model.encryptText(content!!, method!!))
        }
    }
}